### ISSUE TITLE
map_broken

### ISSUE THREAD

**Issue by maxteufel** [tokiwa-software/fuzion#935]
<https://github.com/tokiwa-software/fuzion/blob/7c4604a8a1e12225ae02acd49d3751fab48b5298/lib/stream.fz#L221-L223>

re-discovered with `make check_examples_c` in flang.dev

**Comment by fridis** [comment]
Here is a small example to reproduce this outside the base lib:
```
inherit_outer is

  eimer(T type) ref is

    v T is abstract

    map(B type, f T -> B) : eimer B is
      v B is f eimer.this.v

  bucket(U type, v U) : eimer U is

  hellos => bucket "hello"
  fives => hellos.map s->s.byte_length

  say hellos.v
  say fives.v
```
works interpreted
```
 > ./build/bin/fz inherit_outer.fz
hello
5
```
but not compiled 
```
 > ./build/bin/fz -c inherit_outer.fz

/home/fridi/fuzion/clean/fuzion.2/inherit_outer.fz:8:14: error 1: NYI: in ((inherit_outer.bucket String).map i32).v no targets for Call to (ref Function i32 i32).call target boxed(inherit_outer.fives.ref #fun0):inherit_outer.fives.#fun0
      v B is f eimer.this.v
-------------^

one error.

```

**Comment by fridis** [comment]
The problem seems to be as follows: 

- `UnresolvedType.resolveFeature` first performs `findGenerics`, which searches for the type in the type parameters of all outer features, and only if that failes uses `SourceModule.lookupType`.
- consequently, for a type that matches a type feature in an outer feature and also matches an inherited one, the outer one is found first and used. 
- However, in `AbstractType.applyTypePars_()`, first `AbstractType.applyTypePars(f,actualGenerics)`  is called, which recursively searches parents features, before the outer features are applied. 
- The consequence is, that in the example above, the `T` used in the function type is not replaced by the outer `T` but by the inherited `T`, causing this bug.

Unfortunately, this is not all, changing the order in `applyTypePars` causes new failures.

**Comment by michaellilltokiwa** [comment]
fixed by #2243
